public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -383,6 +384,21 @@ const GrafanaGridItem = React.forwardRef<HTMLDivElement, GrafanaGridItemProps>((
   let width = 100;
   let height = 100;
 
+  const pinnedCount = useRef(0);
+
+  const incrPinnedCount = useCallback((count: number) => {
+    let prevPinnedCount = pinnedCount.current;
+    let nextPinnedCount = (pinnedCount.current = prevPinnedCount + count);
+
+    // redraw with popped z-index when non-zero pinned items
+    if ((prevPinnedCount === 0 && nextPinnedCount > 0) || (prevPinnedCount > 0 && nextPinnedCount === 0)) {
+      setContextVal({ incrPinnedCount });
+    }
+  }, []);
+
+  // here to force redraw via context re-wrap
+  const [contextVal, setContextVal] = useState<LayoutItemContextProps>({ incrPinnedCount });
+
```

Comment:
don't we need both increment and decrement functions? 

If the context only contains increment and decrement we never need to update the context state, that can only live here in a useState (instead of a useRef). 

I find it kind of strange that we update the context with just a new ref to the same callback function

Reply:
> don't we need both increment and decrement functions?

it just does +=, so it handles both. it can be called "adjust" or smth. not sure we need two different fns for this, but happy to change.

> If the context only contains increment and decrement we never need to update the context state, that can only live here in a useState (instead of a useRef).

i wanted to make sure we only redraw during zero/non-zero transitions, not at every counter adjustment, so it does useRef instead of useState and relies on re-creating the context for actual redraw. but maybe there's a cleaner way of doing this?

Reply:
- updated naming `pinnedCount` -> `anchoredCount`
- updated signature to `setAnchoredCount()` that accepts a number or a setter callback that takes current count and returns next count. this should remove any ambiguity.